packages/libp2p/src/connection-manager/index.ts

```diff
@@ -216,8 +218,25 @@ export class DefaultConnectionManager implements ConnectionManager, Startable {
     this.onDisconnect = this.onDisconnect.bind(this)
 
     // allow/deny lists
-    this.allow = (init.allow ?? []).map(ma => multiaddr(ma))
-    this.deny = (init.deny ?? []).map(ma => multiaddr(ma))
+    this.allow = (init.allow ?? []).map((a) => {
+      try {
+        // Try to parse a as a multiaddr directly
+        return convertToIpNet(multiaddr(a))
+      } catch (error) {
+        // If parsing fails, append '/ipcidr/32' and try again
+        return convertToIpNet(multiaddr(a + '/ipcidr/32'))
+      }
```

Comment:
The current could would try to parse the multiaddr twice for the new (recommended) ipcidr multiaddrs. 
I feel like we could do `a.contains('ipcidr')` to either append `ipcidr/32` or not. 

however, there's probably an `.append` or modify multiaddr method.. i'd need to look deeper

Reply:
`.encapsulate` is the method to add to a multiaddr instance

Reply:
@SgtPooki  
Is this ok?

```typescript
// allow/deny lists
    this.allow = (init.allow ?? []).map((a) => {
      try {
        // Attempt to parse `a` with the required /ipcidr/32 if missing
        let ma
        if (a.includes('/ipcidr')) {
          ma = multiaddr(a) // Parse directly if it already includes /ipcidr
        } else {
          ma = multiaddr(a).encapsulate('/ipcidr/32') // Encapsulate with /ipcidr/32 if missing
        }

        return convertToIpNet(ma)
      } catch (error) {
        throw new Error(`Invalid multiaddr format in allow list: ${a}`)
      }
    })

    this.deny = (init.deny ?? []).map((a) => {
      try {
        // Attempt to parse `a` with the required /ipcidr/32 if missing
        let ma
        if (a.includes('/ipcidr')) {
          ma = multiaddr(a) // Parse directly if it already includes /ipcidr
        } else {
          ma = multiaddr(a).encapsulate('/ipcidr/32') // Encapsulate with /ipcidr/32 if missing
        }

        return convertToIpNet(ma)
      } catch (error) {
        throw new Error(`Invalid multiaddr format in deny list: ${a}`)
      }
    })
 ```
 
 
 ########
 
 
 Command failed with exit code 1: npm run dep-check


Missing production dependencies:


@chainsafe/netmask

  /home/runner/work/js-libp2p/js-libp2p/packages/libp2p/src/connection-manager/connection-pruner.ts

  /home/runner/work/js-libp2p/js-libp2p/packages/libp2p/src/connection-manager/index.ts

Do I have to add  "@chainsafe/netmask" ?

git diff

diff --git a/package.json b/package.json

index 886a8d17..17e1f070 100644

--- a/package.json

+++ b/package.json

@@ -52,5 +52,8 @@

   ],

   "overrides": {

     "protons-runtime": "^5.5.0"

+  },

+  "dependencies": {

+    "@chainsafe/netmask": "^2.0.0"

   }

 }



This test error seems unrelated to PR: 
Error: #11 2.514 test/fixtures/relay.ts(20,5): error TS2561: Object literal may only specify known properties, but 'connectionEncryption' does not exist in type 'Libp2pOptions<{ identify: Identify; relay: CircuitRelayService; }>'. Did you mean to write 'connectionEncrypters'?

________________________________
UNRELATED:
Could I work on libp2p http transport example https://github.com/libp2p/js-libp2p/issues/2312

Reply:
> ```ts
> if (a.includes('/ipcidr')) {
> ```

I chatted with @achingbrain about this during Helia WG. Instead of a check on the string, we should construct the multiaddr, `ma = multiaddr(a)` and then check if that instance contains ipcidr with something like [this](https://codepen.io/SgtPooki/pen/LYwgvBe?editors=0011):

```
const ma = multiaddr('/ip4/83.13.55.32')
const maWithCidr = multiaddr('/ip4/83.13.55.32/ipcidr/32')

console.log('ma has ipcidr?', ma.protoNames().includes('ipcidr')) // false
console.log('maWithCidr has ipcidr?', maWithCidr.protoNames().includes('ipcidr')) // true
```

--- 

> Do I have to add  "@chainsafe/netmask" ?

if we directly depend on it, we need to add it to deps in this package.

---

> Could I work on libp2p http transport example https://github.com/libp2p/js-libp2p/issues/2312

I commented on that issue. go for it

Reply:
```typescript
// allow/deny lists
this.allow = (init.allow ?? []).map((a) => {
  try {
    let ma = multiaddr(a) // Attempt to parse the multiaddr

    // Check if the multiaddr already includes '/ipcidr'
    if (!ma.protoNames().includes('ipcidr')) {
      ma = ma.encapsulate('/ipcidr/32') // Encapsulate with /ipcidr/32 if missing
    }

    return convertToIpNet(ma)
  } catch (error) {
    throw new Error(`Invalid multiaddr format in allow list: ${a}`)
  }
})

this.deny = (init.deny ?? []).map((a) => {
  try {
    let ma = multiaddr(a) // Attempt to parse the multiaddr

    // Check if the multiaddr already includes '/ipcidr'
    if (!ma.protoNames().includes('ipcidr')) {
      ma = ma.encapsulate('/ipcidr/32') // Encapsulate with /ipcidr/32 if missing
    }

    return convertToIpNet(ma)
  } catch (error) {
    throw new Error(`Invalid multiaddr format in deny list: ${a}`)
  }
})
```

Reply:
that looks right to me